library/spdm_common_lib/libspdm_com_context_data.c

```diff
@@ -271,20 +276,28 @@ libspdm_return_t libspdm_set_data(void *spdm_context, libspdm_data_type_t data_t
         }
         if (parameter->location == LIBSPDM_DATA_LOCATION_CONNECTION) {
             context->connection_info.capability.ct_exponent = *(uint8_t *)data;
-        } else {
+        } else if (parameter->location == LIBSPDM_DATA_LOCATION_LOCAL) {
             context->local_context.capability.ct_exponent = *(uint8_t *)data;
+        } else {
+            return LIBSPDM_STATUS_INVALID_PARAMETER;
         }
         break;
     case LIBSPDM_DATA_CAPABILITY_RTT_US:
         if (data_size != sizeof(uint64_t)) {
             return LIBSPDM_STATUS_INVALID_PARAMETER;
         }
+        if (parameter->location != LIBSPDM_DATA_LOCATION_LOCAL) {
+            return LIBSPDM_STATUS_INVALID_PARAMETER;
+        }
         context->local_context.capability.rtt = libspdm_read_uint64((const uint8_t *)data);
         break;
     case LIBSPDM_DATA_CAPABILITY_MAX_SPDM_MSG_SIZE:
         if (data_size != sizeof(uint32_t)) {
             return LIBSPDM_STATUS_INVALID_PARAMETER;
         }
+        if (parameter->location != LIBSPDM_DATA_LOCATION_LOCAL) {
```

Comment:
The local `max_spdm_msg_size` is set in https://github.com/DMTF/libspdm/blob/1a554251bdc6566a6cdc8a437617b3c38fe7ac59/include/library/spdm_common_lib.h#L708-L714. Only the connection's `max_spdm_msg_size` should be settable here.

Reply:
Done.